docs/proposals/20241127-namespace-separation.md

```diff
@@ -0,0 +1,297 @@
+---
+title: Multiple Instances of The Same Provider
+authors:
+  - "@marek-veber"
+reviewers:
+  - "@marek-veber"
+creation-date: 2024-11-27
+last-updated: 2025-01-15
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Support running multiple instances of the same provider, each one watching different namespaces 
```

Comment:
if possible, let's keep the title of the PR in sync with the title of the proposal ("namespace separation" is a little bit confusing awhen read without context).

 Also wondering if the title should have something about multi-tenancy or sharding, but I'm not entirely sure at this stage